lexmapr/pipeline_classification.py

```diff
@@ -2,84 +2,222 @@
 
 from inflection import singularize
 from nltk import word_tokenize
+import re
 
 from lexmapr.pipeline_helpers import get_term_parent_hierarchies
+import lexmapr.pipeline_helpers as helpers
 
 
-def refine_ifsac_final_labels(sample, ifsac_final_labels, label_refinements):
-    """TODO..."""
+def customize_order_of_labels(ifsac_final_labels):
+    """Get the final labels in a customized order.
+    :param set ifsac_final_labels: the final labels set to be ordered
+    :rtype: list
+    """
     ret = set(ifsac_final_labels)
+    priority_listing_categories = {"multi-ingredient","veterinary clinical/research",
+                                   "environmental","environmental-water", "environmental-farm",
+                                "environmental-restaurant", "environmental-retail",
+                                "environmental-abattoir", "environmental-warehouse",
+                                "environmental-researchfacility",
+                                "environmental-pasture", "environmental-animal housing",
+                                "environmental-factory/production facility",
+                                "environmental-vehicle", "environmental-construction"}
 
+    revised_final_labels_list = []
+    ifsac_final_labels_list = list(ifsac_final_labels)
+    if ret.intersection(priority_listing_categories):
+        priority_category =  (ret.intersection(priority_listing_categories)).pop()
+        revised_final_labels_list.insert(0, str(priority_category))
+        ifsac_final_labels_list.remove(str(priority_category))
+    for item in sorted(ifsac_final_labels_list):
+        revised_final_labels_list.append(item)
+
+    return revised_final_labels_list
+
+
+def decode_multi_class_labels(ifsac_final_labels):
+    """Decodes the multi class label values in bucket labels.
+    :param set ifsac_final_labels: the final labels set to be ordered
+    :rtype: list
+    """
+    revised_final_labels = set()
+    for label in ifsac_final_labels:
+        if ";" in label:
+            multi_labels = label.split(";")
+            for multi_label in multi_labels:
+                revised_final_labels.add(multi_label)
+        else:
+            revised_final_labels.add(label)
+    revised_final_labels_list = list(revised_final_labels)
+    return revised_final_labels_list
+
+
+def refine_ifsac_final_labels(sample, ifsac_final_labels, label_refinements):
+    """Get the refined final labels after application of customized rules.
+    :param str sample: sample
+    :param set ifsac_final_labels: the final labels set
+    :param dict label_refinements: the dictionary of label refinement resource
+    :rtype: set
+    """
+    # Caution: Rules are sequential - changing the order might change results
```

Comment:
```suggestion
    # Caution: Rules are sequential - changing the order might change 
    # results.
```
Comments should be not exceed 72 chars per line.